### PR TITLE
[CPU] Fix SDPA pattern matching

### PR DESCRIPTION
**Pull request by maxnick**
### Details:
Limit the Concat layer to have maximum 3 children. The third one is allowed to be a ShapeOf op only (to support Mixtral).

### Tickets:
 - 135375

### CODE REVIEW

**Comment by maxnick** [comment]
@luo-cheng2021, could you please review?

File: src/plugins/intel_cpu/src/transformations/cpu_opset/common/pass/stateful_sdpa_fusion.cpp [hunk 1]
```diff
@@ -149,6 +147,28 @@ StatefulSDPAFusion::StatefulSDPAFusion() {
...
         const auto concat_v_node = ov::as_type_ptr<opset6::Concat>(pattern_map.at(concat_v).get_node_shared_ptr());
 
+        for (auto&& item : {concat_k_node, concat_v_node}) {
+            auto&& children = item->get_output_target_inputs(0);
+            switch (children.size()) {
+                case 2:
+                    // pass, as the existence of Assign will be checked later
+                    break;
+                case 3:
+                    // the fist one leads to SDPA, otherwise the matcher doesn't find the pattern
```
Comment (luo-cheng2021) [inline src/plugins/intel_cpu/src/transformations/cpu_opset/common/pass/stateful_sdpa_fusion.cpp:157]: Typo: fist -> first

**Review by luo-cheng2021 — APPROVED** [review]
Just for reminder: if the children of ShapeOf and children's children etc. are not ended at SDPA, there is still faulty fusion. I think it's hardly to avoid all cases, current approach should be acceptable.

**Comment by maxnick** [comment]
> Just for reminder: if the children of ShapeOf and children's children etc. are not ended at SDPA, there is still faulty fusion. I think it's hardly to avoid all cases, current approach should be acceptable.

A correct solution would be a separate matching pass exactly for this Mixtral patter, when there are two branches between SDPA and Concat. And this one is left with the `concat_children_number == 2` condition.

**Pull request merged**